internal/engine/buildcontroller_test.go

```diff
@@ -1201,15 +1201,10 @@ func TestBuildControllerResourceDepTrumpsPendingBuild(t *testing.T) {
 	call := f.nextCall()
 	require.Equal(t, "foo", call.local().Name.String())
 
-	f.b.nextBuildError = errors.New("failure")
 	f.fsWatcher.Events <- watch.NewFileEvent(f.JoinPath("foo", "main.go"))
 	call = f.nextCall()
 	require.Equal(t, "foo", call.local().Name.String())
 
-	f.fsWatcher.Events <- watch.NewFileEvent(f.JoinPath("foo", "main2.go"))
-	call = f.nextCall()
-	require.Equal(t, "foo", call.local().Name.String())
-
```

Comment:
AFAICT, there's not currently a good way to get the fake builder to return errors on successive builds, and between the choices of adding more configuration to the fake builder to allow that or just removing the seemingly useless third build in this test, I went for the latter